src/l500/l500-options.cpp

```diff
@@ -327,17 +338,31 @@ namespace librealsense
                     LOG_INFO("Sensor Mode was changed to: " << sensor_mode_option.get_value_description(rs2_sensor_mode::RS2_SENSOR_MODE_VGA));
                 }
             }
-
-            // TODO!!
-            // If �Max Usable Range� is enabled while camera is not streaming, system will automatically select VGA and Max Range preset 
-            // and not allow changing those settings.!!
-
         }
 
         bool_option::set(value);
     }
 
 
+    void sensor_mode_option::set(float value)
+    {
+        auto &ds = _l500_depth_dev->get_depth_sensor();
+
+        if (ds.supports_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY) && ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY).query() == 1.0f)
+        {
+            ds.get_option(RS2_OPTION_ENABLE_IR_REFLECTIVITY).set(0.0f);
+            LOG_INFO("IR Reflectivity was on - turning it off");
+        }
+
+        if (ds.supports_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE) &&
+            (ds.get_option(RS2_OPTION_ENABLE_MAX_USABLE_RANGE).query() == 1.0) &&
+            (value != rs2_sensor_mode::RS2_SENSOR_MODE_VGA))
+        {
+            throw wrong_api_call_sequence_exception("Max Usable Range support VGA resolution only");
```

Comment:
`supports`

Reply:
I think we should turn it off, not throw

Reply:
👍